Allow routes to be string or pattern.

ncb000gt opened the pull request:
## Description

It's not always advantageous to have the route be a strict string, this allows for the route to be a pattern resolving (#68).

Signed-off-by: Nick Campbell <nicholas.j.campbell@gmail.com>

GiorgiBeriashvili reviewed (approved):
Changes are definitely useful/needed and the accompanying test is a cherry on top.

Approved by me. Thank you!

ncb000gt commented:
The only other thing I can think as being useful would be explicit checks on the types as either String or RegExp, but I'm not sure it's necessary. Open to your thoughts though.

GiorgiBeriashvili commented:
> The only other thing I can think as being useful would be explicit checks on the types as either String or RegExp, but I'm not sure it's necessary. Open to your thoughts though.

It is not necessary, though it certainly is beneficial. Documentation could specify that the expected type is either `String` or `RegExp` as well. Both cases can be handled later when package will undergo a bit of an improvement, though, so it's fine if it's not present now I think.

If you'd like to add explicit checks and maybe hints in the documentation regarding the expected types, it would be awesome!

ncb000gt commented:
I updated the documentation a bit- let me know if that's sufficient for now. :)

ncb000gt commented:
One last update to incorporate the variable naming style we adopted in https://github.com/lomsa-dev/http-mock-adapter/pull/78.

GiorgiBeriashvili commented:
> I updated the documentation a bit- let me know if that's sufficient for now. :)

It sure is for me.

> One last update to incorporate the variable naming style we adopted in #78.

I appreciate your good effort, thank you.

ncb000gt commented:
@GiorgiBeriashvili @campfire5 @LukaGiorgadze I rebased develop (after the merge from the other PR) and fixed the conflicts between the two branches. Those changes have been pushed in 3e77204.
